src/test/groovy/org/gradle/android/SimpleAndroidApp.groovy

```diff
@@ -103,51 +104,108 @@ class SimpleAndroidApp {
 
         file("${library}/build.gradle") << subprojectConfiguration("com.android.library") << activityDependency()
 
+        file("gradle.properties") << """
+                android.useAndroidX=true
+            """.stripIndent()
+
         configureAndroidSdkHome()
     }
 
     private subprojectConfiguration(String androidPlugin) {
         """
             apply plugin: "$androidPlugin"
+            apply plugin: "kotlin-android"
             apply plugin: "org.gradle.android.cache-fix"
 
             repositories {
                 google()
                 jcenter()
             }
 
+            dependencies {
+                def room_version = "2.2.5"
+
+                implementation "androidx.room:room-runtime:\$room_version"
+                annotationProcessor "androidx.room:room-compiler:\$room_version" // For Kotlin use kapt instead of annotationProcessor
```

Comment:
Why don't we use `kapt` here?

Reply:
IIRC, when not using `kapt` or running with `kapt.use.worker.api=false`, setting a relative path for `room.schemaLocation` within the annotationProcessorOptions would work as well.

Reply:
I've added some changes to the room location workaround to handle things for kapt tasks as well (and added tests for it).  Note that there's only so much we can do because of the way kapt handles annotation processor arguments.  For instance, it automatically adds each individual ap argument as a separate input on the task, which means there is no opportunity for us to later prevent that argument from being considered as an input (and mark it as an output directory instead).  So, we change the argument to be relative, which handles the cacheability problem at least.  Right now, it's the best/only option available to us.